roachtest: update multitenant/distsql to use new roachprod service APIs

herkolategan opened the pull request:
Previously the multitenant distsql roachtest relied on an internal util in `roachtest` to start virtual clusters. This PR updates the test to use the new official `roachtest` and `roachprod` APIs for starting virtual clusters.


Some additional changes were required to support upgrading the test. The cluster interface only exposed a method to start storage nodes, but that is insufficient to start virtual clusters that have a separate method on the `roachprod` API (for starting).

This change adds a new method `StartServiceForVirtualCluster` to the cluster interface to enable roachtests to start virtual clusters. Some refactoring was required to enable different sets of cluster settings, depending on what service
type is going to be started.

There are now two sets of cluster settings that can be utilised in `test_runner`. For virtual clusters `virtualClusterSettings` will be used, and for storage clusters `clusterSettings` will be utilised.

Fixes: #116019

Release Note: None
Epic: CRDB-31933

herkolategan committed:
roachtest: add sql instance to functional options

Epic: None
Release Note: None

herkolategan committed:
roachtest: add API to start virtual clusters

Previously the cluster interface only exposed a method to start storage nodes,
but that is insufficient to start virtual clusters that have a separate method
on the `roachprod` API (for starting).

This change adds a new method `StartServiceForVirtualCluster` to the cluster
interface to enable roachtests to start virtual clusters. Some refactoring was
required to enable different sets of cluster settings, depending on what service
type is going to be started.

There are now two sets of cluster settings that can be utilised in
`test_runner`. For virtual clusters `virtualClusterSettings` will be used, and
for storage clusters `clusterSettings` will be utilised.

Epic: None
Release Note: None

herkolategan committed:
roachtest: update multitenant/distsql to use new roachprod service APIs

Previously the multitenant distsql roachtest relied on an internal util in
`roachtest` to start virtual clusters. This change updates the test to use the
new official `roachtest` and `roachprod` APIs for starting virtual clusters.

Fixes: #116019

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115599)
<!-- Reviewable:end -->

herkolategan committed:
roachtest: remove unused functions from multitenant utils

Epic: None
Release Note: None

renatolabs reviewed (commented):
Very excited about this change! :tada: 

> Reviewed 1 of 1 files at r1, 4 of 4 files at r2, 1 of 1 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-:-NlPEi9l04xCHMPSJ27l:b-i4pvex)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan and @srosenberg)

___
*[`pkg/cmd/roachtest/cluster.go` line 1929 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlP7pzP-DrojL5dz-EP:-NlP7pzP-DrojL5dz-EQ:bboup3u) ([raw file](https://github.com/cockroachdb/cockroach/blob/c689b1d6d59ef0858269f02582dbd677298e3e20/pkg/cmd/roachtest/cluster.go#L1929)):*
> ```Go
> 
> func (c *clusterImpl) configureClusterSettingOptions(
> 	defaultEnv map[string]interface{},
> ```

What's the rationale for this `defaultEnv` parameter? From what I can see, we only use it for `COCKROACH_CRASH_ON_SPAN_USE_AFTER_FINISH`. Is there a reason why we wouldn't want that variable in SQL processes? I'm wondering if we should enforce all cockroach processes to have the same default variables (just like we're doing for the random seed var) and eliminate this argument.
___
*[`pkg/cmd/roachtest/cluster.go` line 2031 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlP8NUy9PuS9YyquD_c:-NlP8NUy9PuS9YyquD_d:b-azn6x8) ([raw file](https://github.com/cockroachdb/cockroach/blob/c689b1d6d59ef0858269f02582dbd677298e3e20/pkg/cmd/roachtest/cluster.go#L2031)):*
> ```Go
> 
> 	if settings.Secure {
> 		if err := c.RefetchCertsFromNode(ctx, 1); err != nil {
> ```

Should this really be `1` and not first node in `c.MakeNodes(opts...)`?
___
*[`pkg/cmd/roachtest/cluster.go` line 2011 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPEWsU7QqI-Yu8ozdg:-NlPEWsU7QqI-Yu8ozdh:b-egx09x) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/cluster.go#L2011)):*
> ```Go
> }
> 
> func (c *clusterImpl) StartServiceForVirtualClusterE(
> ```

This function could use some documentation, especially about `externalNodes` and in-memory vs external process tenants.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 71 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPD11bA39qJ1YBPoiW:-NlPD11bA39qJ1YBPoiX:by1tb6p) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L71)):*
> ```Go
> 	c.Start(ctx, t.L(), option.DefaultStartOpts(), settings, c.Node(2))
> 	c.Start(ctx, t.L(), option.DefaultStartOpts(), settings, c.Node(3))
> 	storNodes := c.Range(1, 3)
> ```

Did you mean `storageNodes`?
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 81 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDI-VE6SKvPAchmdQ:-NlPDI-W8xZ4xL1EDpMp:b-9ojcf3) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L81)):*
> ```Go
> 		instStartOps.RoachprodOpts.Target = install.StartServiceForVirtualCluster
> 		instStartOps.RoachprodOpts.VirtualClusterName = tenantName
> 		instStartOps.RoachprodOpts.SQLInstance = sqlInstance
> ```

Interesting, we should probably find a way to package these into something higher level too.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 85 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDR0j8awhkbDNSbZ2:-NlPDR0k6o_y9mIY4pD0:b-7gaj1k) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L85)):*
> ```Go
> 		t.L().Printf("Starting instance %d on node %d", i, node)
> 		instStartOps.RoachprodOpts.SQLPort = 0
> 		instStartOps.RoachprodOpts.AdminUIPort = 0
> ```

Should we add a TODO to remove this once #111052 is resolved?
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 86 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDjMZDizLrdu0b_ms:-NlPDjMZDizLrdu0b_mt:bxa9pv2) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L86)):*
> ```Go
> 		instStartOps.RoachprodOpts.SQLPort = 0
> 		instStartOps.RoachprodOpts.AdminUIPort = 0
> 		err := c.StartServiceForVirtualClusterE(ctx, t.L(), c.Node(node), instStartOps, settings, storNodes)
> ```

Why not `StartServiceForVirtualCluster`?
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 101 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDtau0ET-BSqX9nD3:-NlPDtau0ET-BSqX9nD4:bhofei7) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L101)):*
> ```Go
> 	m := c.NewMonitor(ctx, c.Nodes(1, 2, 3))
> 
> 	//inst1Conn, err := gosql.Open("postgres", c.PGUrl(ctx, 1)
> ```

Delete?
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 146 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPE5IB0dSp3VAFp8hp:-NlPE5IB0dSp3VAFp8hq:b-5927au) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L146)):*
> ```Go
> 		m.Wait()
> 		wg.Done()
> 	}()
> ```

I don't understand this block, why do we need it? I also don't see a `wg.Wait()` call.


<!-- Sent from Reviewable.io -->

srosenberg reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan)

___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 81 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlQ0gY0AF2quEEL9LR_:-NlQ0gY0AF2quEEL9LRa:b-7hi1f4) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/tests/multitenant_distsql.go#L81)):*
> ```Go
> 		instStartOps.RoachprodOpts.Target = install.StartServiceForVirtualCluster
> 		instStartOps.RoachprodOpts.VirtualClusterName = tenantName
> 		instStartOps.RoachprodOpts.SQLInstance = sqlInstance
> ```

Can someone remind me what `SQLInstance` denotes? I can see that it's expected to be `0` in case of `StartSharedProcessForVirtualCluster`, but is exact meaning is somewhat obscured by the implementation of service registry/discovery.


<!-- Sent from Reviewable.io -->

srosenberg reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan)

___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 65 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlQ3ACg5qhtcc-TN4oS:-NlQ3ACg5qhtcc-TN4oT:ba3dfna) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/tests/multitenant_distsql.go#L65)):*
> ```Go
> 	// performance and resource limitations. We set the minimum range max bytes to
> 	// 1 byte to bypass the guardrails.
> 	install.MakeClusterSettings(install.SecureOption(true))
> ```

Remove?
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 81 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlQ0gY0AF2quEEL9LR_:-NlQ2O-vAE6tBmN90RWV:b-e30khk) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/tests/multitenant_distsql.go#L81)):*
Ok, I found the definition in the PR commit message [1] :).

> A SQL Instance can be viewed as a unique identifier for the SQL Instance that is running on a VM in the event that there are more than one SQL Instance serving the same virtual cluster on that VM.

[1] https://github.com/cockroachdb/cockroach/pull/111582


<!-- Sent from Reviewable.io -->

herkolategan committed:
roachtest: document `StartServiceForVirtualClusterE`

Epic: None
Release Note: None

herkolategan committed:
roachtest: add `DefaultStartVirtualClusterOpts` convenience function

Add a convenience function to return `StartOpts` for starting an external
process virtual cluster.

Epic: None
Release Note: None

herkolategan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)

___
*[`pkg/cmd/roachtest/cluster.go` line 1929 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlP7pzP-DrojL5dz-EP:-NlSyw2OEgT7tulfdMhr:bu3oioq) ([raw file](https://github.com/cockroachdb/cockroach/blob/c689b1d6d59ef0858269f02582dbd677298e3e20/pkg/cmd/roachtest/cluster.go#L1929)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

What's the rationale for this `defaultEnv` parameter? From what I can see, we only use it for `COCKROACH_CRASH_ON_SPAN_USE_AFTER_FINISH`. Is there a reason why we wouldn't want that variable in SQL processes? I'm wondering if we should enforce all cockroach processes to have the same default variables (just like we're doing for the random seed var) and eliminate this argument.
</blockquote></details>

Good point, removed `defaultEnv` and made the defaults common to both.
___
*[`pkg/cmd/roachtest/cluster.go` line 2031 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlP8NUy9PuS9YyquD_c:-NlSPfXqE9X-cFq0Nrj-:bxzer87) ([raw file](https://github.com/cockroachdb/cockroach/blob/c689b1d6d59ef0858269f02582dbd677298e3e20/pkg/cmd/roachtest/cluster.go#L2031)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Should this really be `1` and not first node in `c.MakeNodes(opts...)`?
</blockquote></details>

Copied this code over from `StartE`, but I think there is an implicit contract (in `roachprod`) that the absolute first node of any cluster will always carry the certs. Thus after the nodes are started, or any subset of nodes are started, node 1 will contain the latest and will need to be refetched for `roachtest` to operate from locally.
___
*[`pkg/cmd/roachtest/cluster.go` line 2011 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPEWsU7QqI-Yu8ozdg:-NlSPsu18XsYBCs14Ubr:b36cndu) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/cluster.go#L2011)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

This function could use some documentation, especially about `externalNodes` and in-memory vs external process tenants.
</blockquote></details>

Good point, I'll add some docs.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 71 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPD11bA39qJ1YBPoiW:-NlTKNS29jXtLtGQqN_X:bda5hzm) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L71)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Did you mean `storageNodes`?
</blockquote></details>

Naming is from before, but I like the more explicit `storageNodes`, will update.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 81 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDI-VE6SKvPAchmdQ:-NlT6LqpEYltXXbEVX-o:bnp3hbv) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L81)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Interesting, we should probably find a way to package these into something higher level too.
</blockquote></details>

True, I can add a convenience function for service start opts.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 85 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDR0j8awhkbDNSbZ2:-NlTKXtr2v-GlJXjVlOz:b-fznogt) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L85)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Should we add a TODO to remove this once #111052 is resolved?
</blockquote></details>

Good idea, done.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 86 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDjMZDizLrdu0b_ms:-NlT6wHECxCBrfGCJ8Jb:bifevg0) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L86)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Why not `StartServiceForVirtualCluster`?
</blockquote></details>

Opted to try and keep the test similar to how it was with the same logging intact as previously, but can switch if it's cleaner.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 101 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPDtau0ET-BSqX9nD3:-NlT79ckBeXeOSz0HB3c:bcpulx3) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L101)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Delete?
</blockquote></details>

yep, leftover refactoring comment. good catch.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 146 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlPE5IB0dSp3VAFp8hp:-NlSPGWP1bz7w4DrjlHu:b4uks0k) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a0b01dbeb2fd8e3a7ea8d77ba4b30faa29711a6/pkg/cmd/roachtest/tests/multitenant_distsql.go#L146)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

I don't understand this block, why do we need it? I also don't see a `wg.Wait()` call.
</blockquote></details>

Good catch, I did forget the wait; not that it helped unfortunately. This was introduced as a workaround because of a bug I intend to log an issue for, but just realised the workaround also does not work, because I forgot the wait group wait. So I'll remove all of this and log the bug for how this gets stuck on local clusters.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 65 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlQ3ACg5qhtcc-TN4oS:-NlSPCaw7Le8BaomZlYH:b-qh06r9) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/tests/multitenant_distsql.go#L65)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Remove?
</blockquote></details>

Oops, yip my eyes glanced over that one.
___
*[`pkg/cmd/roachtest/tests/multitenant_distsql.go` line 81 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-NlQ0gY0AF2quEEL9LR_:-NlT6Z1l90jfPfV26_Wk:b-qohs6q) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd14aa18eec6179135ae24439765c82a9cb64180/pkg/cmd/roachtest/tests/multitenant_distsql.go#L81)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Ok, I found the definition in the PR commit message [1] :).

> A SQL Instance can be viewed as a unique identifier for the SQL Instance that is running on a VM in the event that there are more than one SQL Instance serving the same virtual cluster on that VM.

[1] https://github.com/cockroachdb/cockroach/pull/111582
</blockquote></details>

Yes, not the clearest of terms, but denotes the ith SQL instance of an external process tenant on a specific VM.


<!-- Sent from Reviewable.io -->

herkolategan reviewed (commented):
Thanks! Updated a few things from comments, and for quickref latest distsql only TC run just for sanity checking (passed): https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13101830

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-:-NlWwEUR4vSSG4ycLneI:b-j9fq6l)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)



<!-- Sent from Reviewable.io -->

renatolabs reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r5, 4 of 4 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-:-Nl_SLeW75gw7KUJz6Mq:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @herkolategan and @srosenberg)

___
*[`pkg/cmd/roachtest/option/options.go` line 61 at r14](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-Nl_S6LL72n9Ji3V86R3:-Nl_S6LL72n9Ji3V86R4:b-na36t9) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b14799b6a40238234cd378a048dd6488c37d34e/pkg/cmd/roachtest/option/options.go#L61)):*
> ```Go
> // DefaultStartVirtualClusterOpts returns StartOpts for starting an external
> // process virtual cluster with the given tenant name and SQL instance.
> func DefaultStartVirtualClusterOpts(tenantName string, sqlInstance int) StartOpts {
> ```

Should we take the `Target` as argument? Or at least make it explicit in the function name that this refers to external process deployment?
___
*[`pkg/cmd/roachtest/option/options.go` line 71 at r14](https://reviewable.io/reviews/cockroachdb/cockroach/115599#-Nl_SFrJ-DRwfhPDmV2u:-Nl_SFrJ-DRwfhPDmV2v:b-3earpp) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b14799b6a40238234cd378a048dd6488c37d34e/pkg/cmd/roachtest/option/options.go#L71)):*
> ```Go
> 	// TODO(herko): remove this once dynamic port assignment is the default.
> 	startOpts.RoachprodOpts.SQLPort = 0
> 	startOpts.RoachprodOpts.AdminUIPort = 0
> ```

The issue was closed today, we should be able to delete this now.


<!-- Sent from Reviewable.io -->

herkolategan commented:
TFTRs!

bors r=renatolabs,srosenberg
